### DISCUSSION TITLE
Select Modules to Compile

### DISCUSSION THREAD

**Discussion by danimasa** [micropython/micropython#9323] [Core Development] [answered]
Hello

How do I select which modules to compile together with Micropython core in some port?
I'm trying to compile the esp32 port but some modules I don't want to use and I'm running out of space in flash, how could I remove the modules I don't need?

**Comment by robert-hh** [comment]
Look into ports/esp32/mpconfigport.h or py/mpconfig.h. There are #defines for enabling or disabling modules. You may also have a look at boards/manifest.py for python modules included into the build as frozen modules.

**Comment by jimmo** [comment, marked as answer]
It depends on the module, but most things are controlled by options in mpconfigport.h and mpconfigboard.h.

You can use this to disable entire modules, specific features from modules, or VM/runtime/language features.

See py/mpconfig.h for the full list.

You can also modify the manifest.py files to remove frozen modules.

**Reply by robert-hh** [reply]
@jimmo: As a unrelated note: You enhanced the manifest.py mechanism recently, but the documentation requires an update. You could also explain the various constants like $MPY_DIR, $BOARD_DIR, etc. ...

**Reply by jimmo** [reply]
Yes absolutely. I haven't finished all the work yet though, and the documentation will follow.

**Reply by danimasa** [reply]
Thanks, correcting the mpconfigport.h already resolved my problem.
